apps/backend/api/graphql/makeSubscriptions.js

```diff
@@ -0,0 +1,79 @@
+import { pipe } from '@graphql-yoga/node'
+import { get } from 'lodash/fp'
+
+// From WebSockets
+// const validMessageTypes = [
+//   x'commentAdded',
+//   x'messageAdded',
+//   x'userTyping',
+//   'newThread',
+//   'newNotification',
+//   'newPost'
+// ]
+
+/**
+ * Filters out subscription events where the current user is the creator.
+ * @param {Object} options - Options for filtering.
+ * @param {Object} options.context - The GraphQL context containing `currentUserId`.
+ * @param {Function} [options.getter] - Optional custom getter to extract the creator ID.
+ * @returns {Function} - A function that filters subscription payloads.
+ */
+const withDontSendToCreator = ({ context, getter } = {}) => {
```

Comment:
Mmm, does this create duplicate records or something? Or is the assumption that the originator will invalidate their cache and cause a refresh anyway?

Reply:
No in the case of an URQL client with graph cache it dedupes by default, just best practice to not send back events to the client that don't belong. The mutation return is how the creator gets updated.

Reply:
Please do keep perusing and such, so very welcome of course, but just to be clear this PR is not ready for any sort of review quite yet. I'll move it out of draft and assign again once it is. Also quite a lot more still changing / to be pushed on this one.